receiver/oracledbreceiver/scraper.go

```diff
@@ -828,3 +828,43 @@ func (s *oracleScraper) shutdown(_ context.Context) error {
 	}
 	return s.db.Close()
 }
+
+func (s *oracleScraper) setupResourceBuilder(rb *metadata.ResourceBuilder) *metadata.ResourceBuilder {
+	rb.SetOracledbInstanceName(s.instanceName)
+	rb.SetHostName(s.hostName)
+	rb.SetServiceInstanceID(s.serviceInstanceID)
+	return rb
+}
+
+func getInstanceID(instanceString string, logger *zap.Logger) string {
+	parts := strings.SplitN(instanceString, "/", 2)
+	hostAndPort := parts[0]
+	service := ""
+	if len(parts) == 2 {
+		service = "/" + parts[1]
+	}
+
+	host, port, err := net.SplitHostPort(hostAndPort)
+	if err != nil {
+		return handleInstanceIDErr(err, logger, port, service)
```

Comment:
I think the flow of logic would make a lot more sense if we simply log the warning message here, and then rename the other method something like `getDefaultServiceInstanceID` or something like that. It's a bit intermingled right now where it logs an err (that it assumes is not nil) and then does some logic to return a string (which isn't clearly documented in the method name).